compiler/passes/scopeResolve.cpp

```diff
@@ -121,13 +176,22 @@ static void handleReceiverFormals() {
       if (UnresolvedSymExpr* sym = toUnresolvedSymExpr(stmt)) {
         SET_LINENO(fn->_this);
 
-        if (TypeSymbol* ts = toTypeSymbol(lookup(sym->unresolved, sym))) {
+        Symbol* rsym = lookup(sym->unresolved, sym);
+        if (TypeSymbol* ts = toTypeSymbol(rsym)) {
           sym->replace(new SymExpr(ts));
 
           fn->_this->type = ts->type;
           fn->_this->type->methods.add(fn);
 
           AggregateType::setCreationStyle(ts, fn);
+
+        } else if (InterfaceSymbol* isym = toInterfaceSymbol(rsym)) {
+          // It is easy to desugar an interface-as-this-type right here.
```

Comment:
Not really sure this comment will help anybody - I think it's saying that this is a convenient place to do the operation. What would be more valuable would be to describe the operation in other words, e.g.  "Convert `arg: SomeInterface` into the same representation as `arg: T where T implements SomeInterface`"